lib/lz4hc.c

```diff
@@ -885,7 +886,7 @@ char* LZ4_slideInputBufferHC(void* LZ4HC_Data)
 {
     LZ4HC_CCtx_internal* const hc4 = &((LZ4_streamHC_t*)LZ4HC_Data)->internal_donotuse;
     int const dictSize = LZ4_saveDictHC((LZ4_streamHC_t*)LZ4HC_Data, (char*)(hc4->inputBuffer), 64 KB);
-    return (char*)(hc4->inputBuffer + dictSize);
+    return (char*)(hc4->inputBuffer) + dictSize;
```

Comment:
Should it return a `const char*`, or do we not want to change the API?

Reply:
It must return a non-`const` pointer, since it is the position at which user is supposed to resume filling the buffer with some new input.

It's a pretty poor interface, and that's why it is deprecated.